samza-api/src/main/java/org/apache/samza/job/model/TaskModel.java

```diff
@@ -35,11 +35,23 @@
   private final TaskName taskName;
   private final Set<SystemStreamPartition> systemStreamPartitions;
   private final Partition changelogPartition;
+  private final TaskMode taskMode;
 
+  /**
+   *  Create a TaskModel for an active task with the given taskName, SSPs, and changelogPartition.
+   * @param taskName The desired taskName
+   * @param systemStreamPartitions SSPs assigned to this task.
+   * @param changelogPartition The changelog SSP for this task.
+   */
   public TaskModel(TaskName taskName, Set<SystemStreamPartition> systemStreamPartitions, Partition changelogPartition) {
+    this(taskName, systemStreamPartitions, changelogPartition, TaskMode.Active);
```

Comment:
I think you have to change the TaskModelMixin for this new field addition.

Reply:
done